src/pages/NIHicWebform.js

```diff
@@ -0,0 +1,2704 @@
+    // NOTE: set this to always false for now to submit dataset without consent info
+    // const isTypeOfResearchInvalid = this.isTypeOfResearchInvalid();
+
+
+    return (
+
+      div({ className: 'container' }, [
+        div({ className: 'col-lg-10 col-lg-offset-1 col-md-12 col-sm-12 col-xs-12' }, [
+          div({ className: 'row no-margin' }, [
+            Notification({notificationData: this.state.notificationData}),
+            div({
+              className: ( 'col-lg-12 col-md-12 col-sm-12 ' )
+            }, [
+              PageHeading({
+                id: 'requestApplication', color: 'common',
+                title: 'Extramural Genomic Data Sharing Plan & Institutional Certification',
+                description: 'This integrated GDSP & IC form combines duplicate fields, allows for digital tracking and statistics, and assigns machine-readable GA4GH Data Use Ontology terms to the datasets upon completion!'
+              })
+            ])
+          ]),
+          hr({ className: 'section-separator' }),
+        ]),
+
+        form({ name: 'form', 'noValidate': true }, [
+          div({ id: 'form-views' }, [
+            div({}, [
+              div({ className: 'col-lg-10 col-lg-offset-1 col-md-12 col-sm-12 col-xs-12' }, [
+                fieldset({}, [
+                  h3({ className: 'rp-form-title common-color' }, ['Administrative Information']),
+
+                  div({className: 'form-group'}, [
+                    div(
+                      {className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group'},
+                      [
+                        label({className: 'control-label rp-title-question common-color'}, [
+                          'Principal Investigator Name* ',
+                        ]),
+                      ]),
+                    div(
+                      {className: 'col-lg-12 col-md-12 col-sm-12 col-xs-12 rp-group rp-last-group'},
+                      [
+                        input({
+                          type: 'text',
+                          name: 'datasetName',
+                          id: 'inputName',
+                          maxLength: '256',
+                          value: this.state.datasetData.datasetName,
+                          onChange: this.handleChange,
+                          className: this.showDatasetNameErrors(this.state.datasetData.datasetName, showValidationMessages),
+                          required: true,
+                        }),
+                        span({
+                          className: 'cancel-color required-field-error-span',
+                          isRendered: fp.includes('required-field-error', this.showDatasetNameErrors(this.state.datasetData.datasetName, showValidationMessages))
+                        },
+                        [this.validateDatasetName(this.state.datasetData.datasetName) ? 'Required field' : 'Dataset Name already in use']),
+                      ])
+                  ]),
```

Comment:
Here's an example of something that is confusing because of copy/paste. We have a form field asking for the PI's name.
Then, as input for that, we are calling it `datasetName`. That input has an error handler attached to it, which we don't need because were not validating a dataset name. I'll file a PR to this PR that does as much cleanup as I can.

Reply:
Thanks @rushtong!